arvo/jael/scry.md

```diff
@@ -0,0 +1,222 @@
+
+## Example
+
+```
+> .^(@ud %j /=life=/(scot %p our))
+1
+```
+
+# `%lyfe`
+
+A `%lyfe` scry gets the current unitized `life` (key revision number). It takes a ship as its `path` and the type returned is a `(unit @ud)`.
+
+## Example
+
+```
+> .^((unit @ud) %j /=lyfe=/(scot %p our))
+[~ 1]
+```
+
+# `%rift`
+
+A `%rift` scry gets the current `rift` (continuity number). It takes a ship as its `path` and the type returned is a `@ud`.
+
+## Example
+
+```
+.^(@ud %j /=rift=/(scot %p our))
+1
+```
+
+# `%ryft`
+
+A `%ryft`  scry gets the current unitized `rift` (continuity number). It takes a ship as its `path` and the type returned is a `(unit @ud)`.
+
+## Example
+
+```
+> .^((unit @ud) %j /=ryft=/(scot %p our))
+[~ 1]
+```
+
+# `%vein`
+
+A `%vein` scry gets your ship's private key for the specified `life`. It takes a `life` `@ud` as its `path` and the type returned is a `ring`.
+
+## Example
+
+```
+> .^(@uw %j /=vein=/1)
+0w84.0MwlQ.y2Ly9.6HVmH.8SYwo.EvuLC.f5YRw.T2NzD.EHtjZ.gpHZb.J0Pu5.aTGVL.UugSA.EZ~E9.~PODC.cohVD.B1zWj.ZWnJ2
+```
+
+# `%vile`
+
+A `%vile` scry gets your `jam`mmed private boot parameters at your ship's current `life`. It takes no additional arguments in its `path`. The type returned is a `@` which is a `jam`med [$seed](@/docs/arvo/jael/data-types.md#seed).
+
+## Examples
+
+```
+> .^(@ %j /=vile=)
+73.825.716.773.695.891.582.219.653.213.376.682.408.892.852.624.025.387.720.465.884.094.267.975.202.807.896.467.150.282.384.122.104.470.678.155.055.914.950.319.747.613.107.324.566.157.366.237.078.063.363.527.599.682.750.233
+```
+
+```
+> ;;(seed:jael (cue .^(@ %j /=vile=)))
+[ who=~zod
+  lyf=1
+    key
+  1.729.646.917.183.337.262.068.568.133.450.460.269.618.308.934.734.494.661.340.450.478.360.301.077.532.415.587.141.242.844.893.269.500.211.196.916.184.768.225.437.577.083.064.475.012.067.992.317.942.521.494.338
+  sig=~
+]
+```
+
+# `%deed`
+
+A `%deed` scry gets the `life`, pubkey and maybe a signature if the ship in question is a comet. It takes the target ship and life as its `path` like `/~sampel-palnet/1` and the type returned is a `[life pass (unit @ux)]`.
+
+## Example
+
+```
+> .^([life pass (unit @ux)] %j /=deed=/(scot %p our)/1)
+[ 1
+  2.649.818.598.466.464.524.534.996.841.661.372.043.001.125.438.577.869.575.450.096.229.472.239.369.342.882.589.479.483.389.735.480.760.982.635.242.955.515.616.049.650.771.700.602.823.946.406.713.457.849.302.626
+  ~
+]
+```
+
+# `%earl`
+
+A `%earl` scry deterministically derives a private key for a moon. Note the `|moon` generator doesn't use this and instead generates moon private keys non-deterministically. It takes the moon name and your ship's current `life` (not the moon's `life`) as its `path` like `/~doznec-dozzod-dozzod-dozzod/1`. The type returned is a [$seed](@/docs/arvo/jael/data-types.md#seed), the `life` of the moon in the `lyf` field of the `seed` will always be `1`.
```

Comment:
So is this just holdover from previous-generation moon infrastructure? cc @philipcmonk

Reply:
Yes